torch/distributed/_tensor/__init__.py

```diff
@@ -20,6 +22,86 @@
 ]
 
 
+def ones(
+    *size,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+    layout: torch.layout = torch.strided,
+    requires_grad: bool = False,
+    device_mesh: Optional[DeviceMesh] = None,
+    placements: Optional[Sequence[Placement]] = None,
+) -> DTensor:
+    """
+    Returns a :class:`DTensor` filled with the scalar value 1, with the shape defined
+    by the variable argument ``size``.
+
+    Args:
+        size (int...): a sequence of integers defining the shape of the output DTensor.
+        Can be a variable number of arguments or a collection like a list or tuple.
+        E.g.: ones(1,2,3..) or ones([1,2,3..]) or ones((1,2,3..))
+
+    Keyword args:
+        dtype (:class:`torch.dtype`, optional): the desired data type of returned DTensor.
+            Default: if ``None``, uses a global default (see :func:`torch.set_default_tensor_type`).
+        device (:class:`torch.device`, optional): the desired device of returned DTensor. 
+            Default: if ``None``, uses the current device for the default tensor type
+            (see :func:`torch.set_default_tensor_type`). device will be the CPU for CPU
+            tensor types and the current CUDA device for CUDA tensor types.
+        layout (:class:`torch.layout`, optional): the desired layout of returned DTensor.
+            Default: ``torch.strided``.
+        requires_grad (bool, optional): If autograd should record operations on the
+            returned tensor. Default: ``False``.
+        device_mesh: :class:`DeviceMesh` type, contains the mesh info of ranks
+        placement: a sequence of :class:`Placement` type: Shard, Replicate, _Partial
+
+    Returns:
+        A :class:`DTensor` object on each rank
+    """
+    # get default device mesh if there's nothing specified
+    device_mesh = get_global_device_mesh() if device_mesh is None else device_mesh
```

Comment:
quick comment: I noticed that a lot of the code is sharing with `zeros`, except the real `torch.ones` call. Can we extract the common logic into a private method, sth like `_common_constructor/initialize` and calls into the common method in all tensor constructors?

Reply:
that makes sense. will do it!